### PR TITLE
fix: replace hardcoded base url protocol to use client-side protocol

### PR DESCRIPTION
**Pull request by BobbieGoede**
Fixes: #666

### CODE REVIEW

**Comment by pd93** [comment]
Also ran into this problem today. I'm running the emulator in a local MiniKube cluster behind TLS and I get a bunch of HTTPS/WSS errors. Is there anything blocking this PR?

File: src/components/Database/DatabaseContainer.tsx [hunk 1]
```diff
@@ -115,7 +115,7 @@ async function fetchDatabases(
   primary: string
 ): Promise<string[]> {
   const res = await fetch(
-    `http://${config.hostAndPort}/.inspect/databases.json?ns=${primary}`
+    `${document.location.protocol}//${config.hostAndPort}/.inspect/databases.json?ns=${primary}`
```
Comment (yuchenshi) [inline src/components/Database/DatabaseContainer.tsx:118]:
```suggestion
    `//${config.hostAndPort}/.inspect/databases.json?ns=${primary}`
```

File: src/components/Database/api.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export class DatabaseApi extends RestApi {
     const [database, { cleanup }] = initDatabase(config, this.namespace);
     this.database = database;
     this.cleanup = cleanup;
-    this.baseUrl = `http://${config.hostAndPort}`;
+    this.baseUrl = `${document.location.protocol}//${config.hostAndPort}`;
```
Comment (yuchenshi) [inline src/components/Database/api.ts:32]:
```suggestion
    this.baseUrl = `//${config.hostAndPort}`;
```

File: src/components/Firestore/FirestoreEmulatedApiProvider.tsx [hunk 3]
```diff
@@ -86,8 +86,8 @@ function useFirestoreRestApi() {
   const databaseId = '(default)';
 
   return {
-    baseUrl: `http://${config.hostAndPort}/v1/projects/${projectId}/databases/${databaseId}`,
-    baseEmulatorUrl: `http://${config.hostAndPort}/emulator/v1/projects/${projectId}/databases/${databaseId}`,
+    baseUrl: `${document.location.protocol}//${config.hostAndPort}/v1/projects/${projectId}/databases/${databaseId}`,
```
Comment (yuchenshi) [inline src/components/Firestore/FirestoreEmulatedApiProvider.tsx:89]:
```suggestion
    baseUrl: `//${config.hostAndPort}/v1/projects/${projectId}/databases/${databaseId}`,
```

File: src/components/Firestore/FirestoreEmulatedApiProvider.tsx [hunk 4]
```diff
@@ -86,8 +86,8 @@ function useFirestoreRestApi() {
...
+    baseEmulatorUrl: `${document.location.protocol}//${config.hostAndPort}/emulator/v1/projects/${projectId}/databases/${databaseId}`,
```
Comment (yuchenshi) [inline src/components/Firestore/FirestoreEmulatedApiProvider.tsx:90]:
```suggestion
    baseEmulatorUrl: `//${config.hostAndPort}/emulator/v1/projects/${projectId}/databases/${databaseId}`,
```

File: src/components/Storage/api/useBuckets.tsx [hunk 5]
```diff
@@ -28,7 +28,9 @@ export function useBuckets() {
   const [bucket] = useBucket();
 
   const fetcher = async () => {
-    const response = await fetch('http://' + config.hostAndPort + '/b');
+    const response = await fetch(
+      `${document.location.protocol}//${config.hostAndPort}/b`
```
Comment (yuchenshi) [inline src/components/Storage/api/useBuckets.tsx:32]:
```suggestion
      `//${config.hostAndPort}/b`
```

File: src/components/Storage/api/useTokens.tsx [hunk 6]
```diff
@@ -25,7 +25,7 @@ export function useTokens(fullPath: string) {
   const key = `storage/tokens/${bucket}/${fullPath}`;
 
   const fetcher = async () => {
-    const url = `http://${
+    const url = `${document.location.protocol}//${
```
Comment (yuchenshi) [inline src/components/Storage/api/useTokens.tsx:28]:
```suggestion
    const url = `//${
```

File: src/components/Storage/api/useTokens.tsx [hunk 7]
```diff
@@ -45,9 +45,9 @@ export function useTokens(fullPath: string) {
 
   async function createToken() {
     await fetch(
-      `http://${config!.hostAndPort}/v0/b/${bucket}/o/${encodeURIComponent(
-        fullPath
-      )}?create_token=true`,
+      `${document.location.protocol}://${
```
Comment (yuchenshi) [inline src/components/Storage/api/useTokens.tsx:48]:
```suggestion
      `//${
```
same, extra `:`

File: src/components/Storage/api/useTokens.tsx [hunk 8]
```diff
@@ -60,9 +60,9 @@ export function useTokens(fullPath: string) {
 
   async function deleteToken(token: string) {
     await fetch(
-      `http://${config!.hostAndPort}/v0/b/${bucket}/o/${encodeURIComponent(
-        fullPath
-      )}?delete_token=${token}`,
+      `${document.location.protocol}://${
```
Comment (yuchenshi) [inline src/components/Storage/api/useTokens.tsx:63]:
```suggestion
      `//${
```

ditto

**Review by yuchenshi — COMMENTED** [review]
LGTM but please apply suggested fixes inline

**Comment by yuchenshi** [comment]
> Also ran into this problem today. I'm running the emulator in a local MiniKube cluster behind TLS and I get a bunch of HTTPS/WSS errors. Is there anything blocking this PR?

This PR is in a great shape, except there are extra colons in a few places. Other places should make use of protocol-relative URLs (mostly code style nits). We'd also appreciate more testing for this broad change and you can definitely help speed this up by patching the suggested changes and testing locally to make sure it works on all tabs (emulators) touched.

**Comment by BobbieGoede** [comment]
Thanks for taking a look and reviewing the PR @yuchenshi, I have applied and committed your suggestions. I haven't had the chance to test all emulators locally yet as my local development setup doesn't use this anymore. Hopefully I will have the time to test it in the coming week.

**Comment by pd93** [comment]
Gave testing this a go today. I managed to load up the UI over https without any issues, but I'm still having some internal connectivity problems. I've logged [a new issue](https://github.com/firebase/firebase-tools-ui/issues/747) so as not to clog up the PR.

**Comment by yuchenshi** [comment]
I'm not 100% sure how, but this PR seems to break Storage unit tests: https://github.com/firebase/firebase-tools-ui/runs/6278540877?check_suite_focus=true#step:5:2864

**Comment by yuchenshi** [comment]
We've made a huge update to the codebase and this PR is outdated. I'm redoing the proposed changes in #858 -- stay tuned and you're welcome to test on that branch to see if that works for you. The goal is to get it merged for real this time.

**Comment by yuchenshi** [comment]
> I'm not 100% sure how, but this PR seems to break Storage unit tests: https://github.com/firebase/firebase-tools-ui/runs/6278540877?check_suite_focus=true#step:5:2864

Found it -- caused by the fetch mocking failing since the URLs no longer start with `http`. Fixing in #858 as well.

**Comment by yuchenshi** [comment]
Released in CLI v11.14.0. It's not perfect, especially not all emulators work over HTTPS, but we think it's a step in the right direction. Let us know if you run into issues and PRs are welcomed!

**Pull request closed without merging**